homeassistant/components/roborock/coordinator.py

```diff
@@ -23,34 +24,38 @@ class RoborockDataUpdateCoordinator(
 ):
     """Class to manage fetching data from the API."""
 
-    def __init__(self, hass: HomeAssistant, client: RoborockMqttClient) -> None:
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        client: RoborockMqttClient,
+        devices_info: dict[str, RoborockHassDeviceInfo],
+    ) -> None:
         """Initialize."""
         super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=SCAN_INTERVAL)
         self.api = client
-        self._devices_prop: dict[str, RoborockDeviceProp] = {}
+        self._devices_info = devices_info
 
     async def release(self) -> None:
         """Disconnect from API."""
         await self.api.async_disconnect()
 
-    async def _update_device_prop(self, device_id: str) -> None:
+    async def _update_device_prop(self, device_info: RoborockHassDeviceInfo) -> None:
         """Update device properties."""
-        device_prop = await self.api.get_prop(device_id)
+        device_prop = await self.api.get_prop(device_info.device.duid)
         if device_prop:
-            if device_id in self._devices_prop:
-                self._devices_prop[device_id].update(device_prop)
+            if device_info.props:
+                device_info.props.update(device_prop)
             else:
-                self._devices_prop[device_id] = device_prop
+                device_info.props = device_prop
 
     async def _async_update_data(self) -> dict[str, RoborockDeviceProp]:
         """Update data via library."""
         try:
-            for device_id in self.api.device_map:
-                await self._update_device_prop(device_id)
-        except RoborockTimeout as ex:
-            raise UpdateFailed(ex) from ex
+            for device_info in self._devices_info.values():
+                await self._update_device_prop(device_info)
         except RoborockException as ex:
             raise UpdateFailed(ex) from ex
-        if self._devices_prop:
-            return self._devices_prop
-        raise UpdateFailed("No device props found")
+        return {
+            device_id: device_info.props
```

Comment:
`device_info.props` can be `None` which seems to conflict with the return type which does not allow `None`.  Is there any value in having a none `RoborockDeviceProp` or should it just always be initialized with empty since the values themselves can still be none?

(Aside: though confusingly the typing on the fields in `RoborockDeviceProp` is typed as if they can't be None, but they can be.  Generally, typing in the library seems wrong in many places)

Reply:
Yeah we need to setup mypy in the package to catch the typing issues. I've fixed this specific issue for now.